src/main/java/com/google/netpcapanalysis/dao/BucketDaoImpl.java

```diff
@@ -0,0 +1,230 @@
+package com.google.netpcapanalysis.dao;
+
+import com.google.netpcapanalysis.models.Flagged;
+import com.google.netpcapanalysis.models.PCAPdata;
+import java.util.*; 
+import java.io.*;
+import java.lang.*;
+
+import io.pkts.PacketHandler;
+import io.pkts.Pcap;
+import io.pkts.buffer.Buffer;
+import io.pkts.packet.Packet;
+import io.pkts.packet.TCPPacket;
+import io.pkts.packet.UDPPacket;
+import io.pkts.packet.IPPacket;
+import io.pkts.protocol.Protocol;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+import com.google.netpcapanalysis.interfaces.dao.PCAPDao;
+import com.google.netpcapanalysis.interfaces.dao.PCAPParserDao;
+import com.google.netpcapanalysis.interfaces.dao.BucketDao;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class BucketDaoImpl implements BucketDao {
+  private ArrayList<PCAPdata> allPCAP; 
+  private ArrayList<PCAPdata> sortedPCAP; 
+  private LinkedHashMap<String, int[]> bucketData = new LinkedHashMap<>();
+  private String myip = "";
+
+  // int[UDP, TCP, IPv4, total]
```

Comment:
Typically instead of an integer Array you would use a HashMap, that way you can add any new protocol you might see without modifying the code, and you also can get rid of the `if` conditions below i.e. instead of `if packet.protocol.equals("UDP")`  it would simply be `classC[packet.protocol.lowercase()]+=1` (in python syntax).

Reply:
Also dont store things for convenience if you can compute them on the fly, e.g. `total` is simply a sum of the previous values right. At best it saves a few clock cycles but this pattern typically results in bugs creeping in